File: Genral_All_Controls/IndicatingOperations/IndicatingOperationsCS/IndicatingOperations/MainForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using Telerik.WinControls.UI;

namespace IndicatingOperations
{
    public partial class MainForm : RadForm
    {
        BackgroundWorker worker;
        DataTable dt;

        public MainForm()
        {
            InitializeComponent();

            worker = new BackgroundWorker();
            worker.DoWork += new DoWorkEventHandler(worker_DoWork);
            worker.RunWorkerCompleted += new RunWorkerCompletedEventHandler(worker_RunWorkerCompleted);

            this.radGridView1.DataBindingComplete += new GridViewBindingCompleteEventHandler(radGridView1_DataBindingComplete);
        }

        private void rbtnPrepareDataSource_Click(object sender, EventArgs e)
        {
            this.rbtnPrepareDataSource.Enabled = false;
            this.radWaitingBar1.Visible = true;
            this.radWaitingBar1.StartWaiting();
            worker.RunWorkerAsync();
        }

        private void rbtnBind_Click(object sender, EventArgs e)
        {
            if (this.radGridView1.DataSource == null)
            {
                WaitingForm.ShowForm();
                this.radGridView1.DataSource = dt;
            }
        }

        void worker_DoWork(object sender, DoWorkEventArgs e)
        {
            dt = new DataTable();

            DataColumn dc = new DataColumn();
            dc.ColumnName = "ID";
            dt.Columns.Add(dc);

            DataColumn dc2 = new DataColumn();
            dc2.ColumnName = "Name";
            dt.Columns.Add(dc2);

            for (int i = 0; i < 1000000; i++)
            {
                DataRow dr = dt.NewRow();
                dr[0] = i;
                dr[1] = "Name " + i.ToString();
                dt.Rows.Add(dr);
            }
        }

        void worker_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {
            this.radWaitingBar1.StopWaiting();
            this.radWaitingBar1.Visible = false;
            this.rbtnBind.Enabled = true;
        }

        void radGridView1_DataBindingComplete(object sender, Telerik.WinControls.UI.GridViewBindingCompleteEventArgs e)
        {
            WaitingForm.CloseForm();
        }
    }
}

File: Genral_All_Controls/IndicatingOperations/IndicatingOperationsVB/IndicatingOperations/MainForm.vb
Imports System.ComponentModel

Public Class MainForm
    Dim WithEvents worker As BackgroundWorker
    Dim dt As DataTable

    Public Sub New()
        ' This call is required by the designer.
        InitializeComponent()
        ' Add any initialization after the InitializeComponent() call.
        worker = New BackgroundWorker()
    End Sub

    Private Sub rbtnPrepareDataSource_Click(sender As System.Object, e As System.EventArgs) Handles rbtnPrepareDataSource.Click
        Me.rbtnPrepareDataSource.Enabled = False
        Me.RadWaitingBar1.Visible = True
        Me.RadWaitingBar1.StartWaiting()
        worker.RunWorkerAsync()
    End Sub

    Private Sub rbtnBind_Click(sender As System.Object, e As System.EventArgs) Handles rbtnBind.Click
        If Me.RadGridView1.DataSource Is Nothing Then
            WaitingForm.ShowForm()
            Me.RadGridView1.DataSource = dt
        End If
    End Sub

    Private Sub worker_DoWork(sender As System.Object, e As DoWorkEventArgs) Handles worker.DoWork
        dt = New DataTable()

        Dim dc As New DataColumn()
        dc.ColumnName = "ID"
        dt.Columns.Add(dc)

        Dim dc2 As New DataColumn()
        dc2.ColumnName = "Name"
        dt.Columns.Add(dc2)

        For i As Integer = 0 To 999999
            Dim dr As DataRow = dt.NewRow()
            dr(0) = i
            dr(1) = "Name " & i.ToString()
            dt.Rows.Add(dr)
        Next i
    End Sub

    Private Sub worker_RunWorkerCompleted(sender As System.Object, e As RunWorkerCompletedEventArgs) Handles worker.RunWorkerCompleted
        Me.RadWaitingBar1.StopWaiting()
        Me.RadWaitingBar1.Visible = False
        Me.rbtnBind.Enabled = True
    End Sub

    Private Sub RadGridView1_DataBindingComplete(sender As Object, e As Telerik.WinControls.UI.GridViewBindingCompleteEventArgs) Handles RadGridView1.DataBindingComplete
        WaitingForm.CloseForm()
    End Sub
End Class
